src/Nethermind/Nethermind.Network.Discovery/DiscoveryApp.cs

```diff
@@ -500,77 +430,91 @@ private async Task<bool> InitializeBootnodes(CancellationToken cancellationToken
         return reachedNodeCounter > 0;
     }
 
-    private void RunDiscoveryProcess()
+    private async Task RunDiscoveryProcess()
     {
-        Task disc = RunDiscoveryAsync(_appShutdownSource.Token).ContinueWith(t =>
+        byte[] randomId = new byte[64];
+        CancellationToken cancellationToken = _appShutdownSource.Token;
+        PeriodicTimer timer = new(TimeSpan.FromMilliseconds(10));
+
+        long lastTickMs = Environment.TickCount64;
+        long waitTimeTimeMs = 10;
+        while (!cancellationToken.IsCancellationRequested
+            && await timer.WaitForNextTickAsync(cancellationToken))
         {
-            if (t.IsFaulted)
+            long currentTickMs = Environment.TickCount64;
+            long elapsedMs = currentTickMs - lastTickMs;
+            if (elapsedMs < waitTimeTimeMs)
             {
-                _logger.Error($"Error during discovery process: {t.Exception}");
+                // Wait for the remaining time
+                await Task.Delay((int)(waitTimeTimeMs - elapsedMs), cancellationToken);
```

Comment:
is this needed? doesn't timer handle that?

Reply:
Ah looks like we can fix it in .net 8: https://github.com/dotnet/runtime/pull/82560, please add TODO or issue